packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2134,12 +2147,22 @@ class TextSelectionGestureDetectorBuilder {
 
     if (!_isShiftTapping) {
       // Adjust the drag start offset for possible viewport offset changes.
-      final Offset startOffset = renderEditable.maxLines == 1
+      final Offset editableOffset = renderEditable.maxLines == 1
           ? Offset(renderEditable.offset.pixels - _dragStartViewportOffset, 0.0)
           : Offset(0.0, renderEditable.offset.pixels - _dragStartViewportOffset);
-
+      final ScrollableState? scrollableState =
+          delegate.editableTextKey.currentContext == null
+              ? null
+              : Scrollable.of(delegate.editableTextKey.currentContext!);
+      final double currentScrollDy = scrollableState == null
+          ? 0.0
+          : scrollableState.position.pixels;
```

Comment:
Can this and the same chunk of lines starting at 2122 be wrapped up in a method?